03-Bash/find_dup_files/find_dup_files.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+function help()
+{
+    echo -e "Usage: find_dup_files [OPTION] [PATH]\nSearches for files with MD5 checksum (128-bit) collisions in the specified with PATH directory (by default in current directory).\n\nOption:\n\t-h, --help\t\tprint this message and exit\n"
+}
+
+function find_dup_files()
+{
+    SEARCH_PATH=$1
+    PREV_FILE_PATH=""
+    PREV_CHECKSUM=""
+
+    if [ "${SEARCH_PATH:0:1}" != "/" ]; then
+        cd $SEARCH_PATH
+        SEARCH_PATH=$(pwd)
+        cd - &>/dev/null
+    fi
+
+    find $SEARCH_PATH -type f -exec md5sum {} \; | sort | \
+        while read -r checksum file_path; do
+            if [ "$PREV_CHECKSUM" = "" ]; then
+                PREV_CHECKSUM=$checksum
+                PREV_FILE_PATH=$file_path
+                continue
+            fi
+            if [ "$PREV_CHECKSUM" = "$checksum" ]; then
+                echo -e "MD5 checksum of file:"
+                echo -e ">>> $file_path"
+                echo -e "with size of $(stat -c%s "$file_path") bytes is equal to MD5 checksum of file:"
+                echo -e ">>> $PREV_FILE_PATH"
+                echo -e "wiht size of $(stat -c%s "$PREV_FILE_PATH") bytes.\n"
+            else
+                PREV_CHECKSUM=$checksum
+                PREV_FILE_PATH=$file_path
+            fi
+        done
+}
+
+if [ $# -gt 1 ]; then
+    echo -e "Incorrect usage!"
+    help
+elif [ $# -eq 1 ]; then
+    arg="$1"
+    case $arg in
+        -h|--help)
+            help
+            ;;
+        -*)
+            echo -e "Incorrect usage!"
```

Comment:
What is the reason for option '-e'?

Reply:
A habit based on the requirements of the project I'm working on: always use the '-e' option to display messages.